### PR TITLE
Add map of (initial_loc, initial_set)

### LINKED ISSUE #283: Wrong approach to initialise waiting list at the very beginning.
Currently we add only `loc1` and `X0` to the waiting list. However, we should add all locations from our system with the same `X0` to the waiting list. In addition, user should have an option to specify location via model file.

### PR DESCRIPTION
**Pull request by kostakoida**
Closes #283.

### CODE REVIEW

File: src/options.jl [hunk 1]
```diff
@@ -245,6 +246,7 @@ function validate_solver_options_and_add_default_values!(options::Options)::Opti
     check_aliases_and_add_default_value!(dict, dict_copy, [:max_jumps], 5)
     check_aliases_and_add_default_value!(dict, dict_copy, [:fixpoint_check], true)
     check_aliases_and_add_default_value!(dict, dict_copy, [:n], nothing)
+    check_aliases_and_add_default_value!(dict, dict_copy, [:init], Tuple{Int64,LazySets.LazySet{Float64}}[])
```
Comment (schillic) [inline src/options.jl]: `Int64` -> `Int` (`Int64` falls back to `Int64` on 64-bit systems, but `Int` is shorter and cleaner)

File: test/Reachability/unit_solve_hybrid_bouncing_ball.jl [hunk 2]
```diff
@@ -31,11 +31,13 @@ HS = HybridSystem(a, m, r, s);
 # initial condition in mode 1
 X0 = Hyperrectangle(low=[10, 0.0], high=[10.2, 0.0]);
 
+X_loc1 = Hyperrectangle(low=[9.8, 0.0], high=[10, 0.0]);
```
Comment (schillic) [inline test/Reachability/unit_solve_hybrid_bouncing_ball.jl]: Why is this set different from `X0`?

File: src/solve.jl [hunk 3]
```diff
@@ -110,7 +110,27 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
     # - (discrete) location
     # - (set of) continuous-time reach sets
     # - number of previous jumps
-    waiting_list = [(1, ReachSet{LazySet{N}, N}(X0, zero(N), zero(N)), 0)]
+    inits = length(options[:init]) > 0 ? options[:init] : [(n, X0) for n in states(HS)]
+
+    waiting_list = []
```
Comment (schillic) [inline src/solve.jl]: Better initialize it with the right type: `Vector{Tuple{Int, ReachSet{LazySet{N}, N}, Int}}()`

File: src/solve.jl [hunk 4]
```diff
@@ -110,7 +110,27 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
...
+    for (modeId, x0) in inits
+        mode = HS.modes[modeId]
+        source_invariant = mode.X
+
+        if source_invariant isa HalfSpace
+            # TODO temporary conversion to HPolytope
+            source_invariant = HPolytope([source_invariant])
+        else
+            @assert source_invariant isa HPolytope
```
Comment (schillic) [inline src/solve.jl]: I would just drop this `else` block. If the intersection fails, we will see an error message.

File: src/solve.jl [hunk 5]
```diff
@@ -110,7 +110,27 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
...
+        end
+
+        loc_x0sets = intersection(source_invariant, x0)
```
Comment (schillic) [inline src/solve.jl]: I find the name `loc_x0sets` confusing. Why the "*s*" in "set*s*"?

File: test/Reachability/unit_solve_hybrid_bouncing_ball.jl [hunk 6]
```diff
@@ -31,11 +31,13 @@ HS = HybridSystem(a, m, r, s);
...
+
 system = InitialValueProblem(HS, X0);
 input_options = Options(:mode=>"reach");
 
-problem_options = Options(:vars=>[1,2], :T=>5.0, :δ=>0.1, :plot_vars=>[1, 2],
-                          :max_jumps=>1, :verbosity=>1);
+problem_options = Options(:vars=>[1,2], :T=>10.0, :δ=>0.1, :plot_vars=>[1, 2],
+                          :max_jumps=>10, :verbosity=>1, :init=>[(1, X_loc1)]);
```
Comment (schillic) [inline test/Reachability/unit_solve_hybrid_bouncing_ball.jl]: Do we need to have that many jumps? I was reducing the time and jumps on purpose to have cheaper tests.

File: test/Reachability/unit_solve_hybrid_bouncing_ball.jl (hunk 2 above)
Comment (kostakoida) [reply test/Reachability/unit_solve_hybrid_bouncing_ball.jl]: just played around to make sure that it applies new X_loc1

File: src/options.jl [hunk 7]
```diff
@@ -197,6 +197,7 @@ Supported options:
 - `:plot_vars`     -- variables for projection and plotting;
                       alias: `:output_variables`
 - `:n`             -- system's dimension
+- `:init`             -- map of (initial_locations, initial_set)
```
Comment (schillic) [inline src/options.jl]: remove again

File: src/solve.jl [hunk 8]
```diff
@@ -90,16 +90,26 @@ Interface to reachability algorithms for a hybrid system PWA dynamics.
...
     opC, opD = default_operator(system)
+    HS = system.s
+    X0 = system.x0
+    inits = [(n, X0) for n in states(HS)]
+    system = InitialValueProblem(HS, inits)
     return solve(system, options, opC, opD)
 end
 
-function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
+function solve(system::InitialValueProblem{<:HybridSystem, <:Array{<:Tuple{Int64,<:LazySets.LazySet{N}},1}},
```
Comment (schillic) [inline src/solve.jl]: Just a remark: `Array{T, 1}` can be conveniently written as `Vector{T}`.

**Comment by kostakoida** [comment]
No, we do not need to change the thermostat model. We have to signatures:
`function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
               options::Options)::AbstractSolution where {N}`
and 
`function solve(system::InitialValueProblem{<:HybridSystem, <:Vector{<:Tuple{Int64,<:LazySets.LazySet{N}}}},
               options::Options)::AbstractSolution where {N}`

**Comment by schillic** [comment]
Okay, we do not need to, but it would be more efficient :smile:

**Pull request merged**